app/src/main/java/com/github/freeman/bootcamp/DrawHatActivity.kt

```diff
@@ -0,0 +1,361 @@
+package com.github.freeman.bootcamp
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.graphics.Bitmap
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.annotation.DrawableRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.asAndroidBitmap
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.net.toUri
+import coil.compose.rememberAsyncImagePainter
+import com.github.freeman.bootcamp.DrawHatActivity.Companion.DRAW_HAT
+import com.github.freeman.bootcamp.DrawHatActivity.Companion.HAT_HELP
+import com.github.freeman.bootcamp.DrawHatActivity.Companion.YOUR_HAT
+import com.github.freeman.bootcamp.games.guessit.drawing.*
+import com.github.freeman.bootcamp.ui.theme.BootcampComposeTheme
+import com.github.freeman.bootcamp.utilities.firebase.FirebaseUtilities
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.StorageReference
+import com.google.firebase.storage.ktx.storage
+import io.ak1.drawbox.DrawBox
+import io.ak1.drawbox.DrawController
+import io.ak1.drawbox.rememberDrawController
+import io.ak1.rangvikalp.RangVikalp
+import java.io.ByteArrayOutputStream
+import java.io.File
+
+class DrawHatActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val storageRef = Firebase.storage.reference
+        setContent {
+            BootcampComposeTheme {
+                DrawHatScreen(storageRef)
+            }
+        }
+    }
+
+    companion object {
+        const val DRAW_HAT = "Draw Your Hat"
+        const val HAT_HELP = "Welcome! Please draw and upload any hat for your in-game video filter" +
+                " by clicking the upload symbol in the right corner."
+        const val YOUR_HAT = "Your hat"
+    }
+}
+
+// Note: DrawingActivity could unfortunately not be reused as it contains references to a specific
+// game, contains a timer, uses the realtime database as opposed to the storage one, etc.
```

Comment:
I see that you had to reuse some code from the DrawingActivity. 
I think it's not a problem to reuse code here as extracting common elements from both activities to make the code more modular would take too much time (only one sprint left with higher priority tasks).

Reply:
You are absolutely right. It's really such a shame that I didn't modularise them, but I did try to do it before realising that I was simply making a mess of the code and thus that in order to do this in a clean way, this would require much more refactoring and hence time.